Lib/test/test_asyncio/test_subprocess.py

```diff
@@ -686,6 +686,87 @@ async def execute():
 
         self.assertIsNone(self.loop.run_until_complete(execute()))
 
+    def test_create_subprocess_env_shell(self) -> None:
+        async def main() -> None:
+            cmd = f'''{sys.executable} -c "import os, sys; sys.stdout.write(os.getenv('FOO'))"'''
+            env = {"FOO": 'bar'}
+            proc = await asyncio.create_subprocess_shell(
+                cmd, env=env, stdout=subprocess.PIPE
+            )
+            stdout, _ = await proc.communicate()
+            self.assertEqual(stdout, b"bar")
+            self.assertEqual(proc.returncode, 0)
+            task = asyncio.create_task(proc.wait())
+            await asyncio.sleep(0)
+            self.assertEqual(task.result(), proc.returncode)
```

Comment:
Why not `await task` instead of `sleep(0)` and `task.result()`? At this point we already know the process has exited, since the returncode is 0 on the previous line.

Reply:
> At this point we already know the process has exited, since the returncode is 0 on the previous line.

Yes, awaiting can take more than one cycle but we are checking that wait will be done immediately so using `sleep(0)` trick.